engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionCallbacks.java

```diff
@@ -45,55 +59,98 @@ final class ExecutionCallbacks implements IdExecutionService.Callbacks {
    * @param onCachedCallback the consumer of the cached value events.
    */
   ExecutionCallbacks(
-          UUID nextExecutionItem,
-          RuntimeCache cache, MethodCallsCache methodCallsCache, UpdatesSynchronizationState syncState,
-          Consumer<ExpressionValue> onCachedCallback, Consumer<ExpressionValue> onComputedCallback,
-          Consumer<ExpressionCall> functionCallCallback
-  ) {
+      VisualizationHolder visualizationHolder,
+      UUID nextExecutionItem,
+      RuntimeCache cache,
+      MethodCallsCache methodCallsCache,
+      UpdatesSynchronizationState syncState,
+      Consumer<ExpressionValue> onCachedCallback,
+      Consumer<ExpressionValue> onComputedCallback,
+      Consumer<ExpressionCall> functionCallCallback,
+      Consumer<ExecutedVisualization> onExecutedVisualizationCallback) {
+    this.visualizationHolder = visualizationHolder;
     this.nextExecutionItem = nextExecutionItem;
     this.cache = cache;
     this.methodCallsCache = methodCallsCache;
     this.syncState = syncState;
     this.onCachedCallback = onCachedCallback;
     this.onComputedCallback = onComputedCallback;
     this.functionCallCallback = functionCallCallback;
+    this.onExecutedVisualizationCallback = onExecutedVisualizationCallback;
   }
 
+  @Override
+  public void onEnter(UUID nodeId) {}
+
+  @Override
+  public void onExpressionReturn(UUID nodeId, Object result, long nanoElapsedTime) {}
+
   @CompilerDirectives.TruffleBoundary
-  public final Object findCachedResult(UUID nodeId) {
```

Comment:
Why are we removing `findCachedResult`? It worked well as far as I can tell.

Reply:
It's there. I just moved methods around.